app/controllers/v1/users_controller.rb

```diff
@@ -38,20 +55,35 @@ def exchange_login_code
       login_code = params[:login_code]
 
       return render_not_found unless user
+      
+      if !login_code.nil?
+        if user.auth_type == 'totp'
+          if ROTP::TOTP.new(user.totp_key).verify(login_code, drift_behind: 15)
```

Comment:
Generally bad practice to use magic constants, like `15`. Refactor out into its own variable w a descriptive name